### PR TITLE
feat: add ipc set title window event

### PR DESCRIPTION
**Pull request by cyrinux**
Hi, 

This is a very basic PR to be able to get an event only when a window title change. (active and inactive windows)

This will allow me to refresh https://github.com/hyprland-community/hyprland-autoname-workspaces on title update.

Also, hyprland-rs PR is ready for this: https://github.com/hyprland-community/hyprland-rs/pull/56/files

I don't understand/know if we need also to add a `EMIT_HOOK_EVENT("windowTitle", PWINDOW);`

Should I ?

### CODE REVIEW

File: src/events/Windows.cpp [hunk 1]
```diff
@@ -805,6 +805,7 @@ void Events::listener_setTitleWindow(void* owner, void* data) {
         return;
 
     PWINDOW->m_szTitle = g_pXWaylandManager->getTitle(PWINDOW);
+    g_pEventManager->postEvent(SHyprIPCEvent{"settitlewindow", getFormat("%lx", PWINDOW)});
```
Comment (vaxerski) [inline src/events/Windows.cpp:808]: bad name, doesn't fit the rest of the events. I'd prefer `windowtitle`.

**Review by vaxerski — COMMENTED** [review]
make a wiki mr too pls

**Comment by vaxerski** [comment]
> I don't understand/know if we need also to add a EMIT_HOOK_EVENT("setTitleWindow", PWINDOW);

No _need_ but it would be nice

**Comment by cyrinux** [comment]
Thanks @vaxerski,
I change the name, add the HOOK, and create wiki PR here: https://github.com/hyprwm/hyprland-wiki/pull/239

**Review by vaxerski — APPROVED** [review]
gangsta

**Comment by cyrinux** [comment]
🍰

**Pull request merged**